packages/web-config-server/src/apiV1/dataBuilders/helpers/translateEventEntityIdsToNames.js

```diff
@@ -8,14 +8,20 @@ import { transformValue } from 'apiV1/dataBuilders/transform';
 export const translateEventEntityIdsToNames = async (models, events, dataElementCodes) => {
   return Promise.all(
     events.map(async event => {
-      const updatedDataValues = await dataElementCodes.reduce(async (acc, dataElementCode) => {
-        const dataValueEntityId = event.dataValues[dataElementCode];
-        if (dataValueEntityId) {
-          const entityName = await transformValue(models, 'entityIdToName', dataValueEntityId);
-          return { ...acc, [dataElementCode]: entityName };
-        }
-        return acc;
-      }, {});
+      const updatedDataValues = await dataElementCodes.reduce(
+        async (accPromise, dataElementCode) => {
+          // Since this is an async function after the first loop returns, the accumulator is a promise
+          // So await the promise before continuing
+          const acc = await accPromise;
+          const dataValueEntityId = event.dataValues[dataElementCode];
+          if (dataValueEntityId) {
+            const entityName = await transformValue(models, 'entityIdToName', dataValueEntityId);
+            return { ...acc, [dataElementCode]: entityName };
+          }
+          return acc;
+        },
+        {},
+      );
```

Comment:
This is cool, and obviously works well, but honestly I think it'd be less confusing to just mutate a variable that's declared above a `.map` of promises (or even a classic `for of` loop for that matter). Javascript functional patterns are all well and good, but sometimes they bury the lede imo
```suggestion
      const updatedDataValues = {};
      await Promise.all(dataElementCodes.map(async dataElementCode => {
        const dataValueEntityId = event.dataValues[dataElementCode];
        if (dataValueEntityId) {
          const entityName = await transformValue(models, 'entityIdToName', dataValueEntityId);
          updatedDataValues[dataElementCode] = entityName;
        }
      }));
```

Reply:
Oh that's a good call. I think when bug-fixing my mind tends to focus on finding the issue in the written code, and I don't consider improvements while I'm there.